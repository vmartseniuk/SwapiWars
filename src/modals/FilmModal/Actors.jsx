/* eslint-disable react/prop-types */
/* eslint-disable react/jsx-one-expression-per-line */
import React, { memo } from "react";
import { Grid, makeStyles } from "@material-ui/core";
import { CardSpinner } from "components/Spinner";

const useStyles = makeStyles(() => ({
  actorsContainer: {
    width: "100%",
    position: "relative",
  },
  text: {
    textAlign: "left",
    width: "inherit",
    fontSize: "20px",
    paddingBottom: "10px",
  },
}));
const Characters = (props) => {
  const classes = useStyles();
  const { character, loading } = props;

  return (
    <div className={classes.actorsContainer}>
      <div className={classes.text}>Characters:</div>
      <CardSpinner loading={loading} />
      <Grid container>
        {character.map((item) => (
          <Grid item xs={3} key={item}>
            {item}
          </Grid>
        ))}
      </Grid>
    </div>
  );
};

export default memo(Characters);
