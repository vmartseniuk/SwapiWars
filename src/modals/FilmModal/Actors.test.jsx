import { create } from "react-test-renderer";
import Actors from "./Actors";

describe("FilmModal/Actors", () => {
  it("renders correctly", () => {
    const arrNames = ["name1", "name2"];
    const rendered = create(<Actors character={arrNames} />);
    expect(rendered.toJSON()).toMatchSnapshot();
  });
});
