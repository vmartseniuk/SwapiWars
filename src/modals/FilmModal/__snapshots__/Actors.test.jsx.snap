// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilmModal/Actors renders correctly 1`] = `
<div
  className="makeStyles-actorsContainer-1"
>
  <div
    className="makeStyles-text-2"
  >
    Characters:
  </div>
  <div
    className="css-n9lk9p"
  >
    <div
      className="css-11c1yvw"
    />
    <div
      className="css-1hhfbxh"
    />
    <div
      className="css-1tz7pd6"
    />
    <div
      className="css-1oe4hzs"
    />
    <div
      className="css-1p48bko"
    />
    <div
      className="css-18pv7bu"
    />
  </div>
  <div
    className="MuiGrid-root MuiGrid-container"
  >
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      name1
    </div>
    <div
      className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
    >
      name2
    </div>
  </div>
</div>
`;
